src/main.rs

```diff
@@ -214,16 +217,15 @@ fn enable_write_protect_bit() {
     unsafe { Cr0::update(|cr0| *cr0 |= Cr0Flags::WRITE_PROTECT) };
 }
 
-#[lang = "panic_fmt"]
+#[panic_implementation]
 #[no_mangle]
-pub extern "C" fn rust_begin_panic(
-    msg: core::fmt::Arguments,
-    file: &'static str,
-    line: u32,
-    _column: u32,
-) -> ! {
+pub extern "C" fn panic(info: &PanicInfo) -> ! {
     use core::fmt::Write;
-    write!(printer::Printer, "PANIC: {} in {}:{}", msg, file, line).unwrap();
+    if let Some(location) = info.location() {
+        write!(printer::Printer, "PANIC: {} in {}:{}", info.message().unwrap(), location.file(), location.line()).unwrap();
+    } else {
+        write!(printer::Printer, "PANIC: {} at ???", info.message().unwrap()).unwrap();
+    }
```

Comment:
PanicInfo implements Display, so I think we can just do `println!("PANIC: {}", info)` or even `println!("{}", info)`. We also wouldn't need `#![feature(panic_info_message)]` then. Or is there some advantage in printing explicitly?

Reply:
Nope, I just missed that `PanicInfo` implemented `Display`. Thanks for the review; fixed!